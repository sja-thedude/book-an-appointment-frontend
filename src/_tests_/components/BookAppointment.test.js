import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../redux/Store';
import BookAppointment from '../../components/dashboard/BookAppointment';

describe('BookAppointment', () => {
  it('renders correctly', () => {
    window.localStorage.setItem('bookDoctorUser', 1);
    const tree = render(<Provider store={store}><Router><BookAppointment /></Router></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
