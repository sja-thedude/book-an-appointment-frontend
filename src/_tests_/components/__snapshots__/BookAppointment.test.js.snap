// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookAppointment renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="book-appointment-section"
      >
        <h2
          class="title"
        >
          Doctor book appointment
        </h2>
        <h4>
          Kindly Go back to doctors list to select your favourite doctor
        </h4>
        <form>
          <div
            class="form-group"
          >
            <h3
              class="sub-title"
            >
              Choose a Specific Day
            </h3>
            <input
              class="form-control"
              id="date"
              placeholder="date"
              type="date"
              value=""
            />
            <input
              class="form-control"
              id="date"
              placeholder="time fro appointment"
              type="time"
              value=""
            />
          </div>
          <div
            class="btn-appointment"
          >
            <button
              type="submit"
            >
              Book Now
            </button>
          </div>
        </form>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="book-appointment-section"
    >
      <h2
        class="title"
      >
        Doctor book appointment
      </h2>
      <h4>
        Kindly Go back to doctors list to select your favourite doctor
      </h4>
      <form>
        <div
          class="form-group"
        >
          <h3
            class="sub-title"
          >
            Choose a Specific Day
          </h3>
          <input
            class="form-control"
            id="date"
            placeholder="date"
            type="date"
            value=""
          />
          <input
            class="form-control"
            id="date"
            placeholder="time fro appointment"
            type="time"
            value=""
          />
        </div>
        <div
          class="btn-appointment"
        >
          <button
            type="submit"
          >
            Book Now
          </button>
        </div>
      </form>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
